### ISSUE TITLE
Provide advice on cloud providers

### ISSUE THREAD

**Issue by mlissner** [jlund/streisand#318]
This is an awesome project and it takes me almost all the way from "I know nothing" to "I have a functional VPN." The only missing piece is that it provides an abundance of cloud provider options without providing advice as to which to choose.

It would be really helpful if there was just a _little_ advice. For example:
- Google - Choose this if you can keep your traffic below 2GB/month.
- DO - Choose this if you expect to not use your VPN for months at a time.
- Linode - Choose this for a provider outside of the US.

Something like that. As it is, I guess I'll have to go shop around and compare all these providers, which is actually a lot of work. For me, I just want something super cheap that I can rely on every month or two when I travel...

**Comment by jlund** [comment]
Interesting idea. I'll have to think about how that would work. I would also like to remain neutral. I sincerely like all of the providers that are natively supported right now :)

For your use case, I would recommend DigitalOcean. The servers that Streisand sets up there are only $5 per month, which is about half the cost of the other providers. Performance is also good.

**Comment by mlissner** [comment]
I'm curious...I set up AWS because I was most familiar with it, but now I see that AWS uses the micro instance with 1GB of RAM and DO uses the $5 model with half a gig of RAM. If we use AWS nano instead, it looks like it will cost about as much as the DO, and have the same amount of RAM.

(Thanks for the advice!)

**Comment by fghorow** [comment]
I just set one up with a VPS on BudgetVM (following the "Running Streisand on Other Providers" instructions). They have a USD25/year OpenVZ thing with about 512M RAM, 1 core, 2TB of traffic/month, and something like 10GB disk. Unfortunately, the L2TP/IPsec connection does not work with OpenVZ (ports showing open in VM, but confirmed closed with nmap from the outside). After a little palavering with the BudgetVM sales team, I upgraded that server to the USD35/year Xen based server with substantially the same specs. (I think the amount of disk might be smaller, but for this application, who cares?)
So, if you're looking for a cheap-o way of running streisand, and L2TP/IPsec is of any utility to you, I can confirm that the same script (with only IP number changed) works under Xen, but fails under OpenVZ (on BudgetVM's configuration, at least).
Good Luck!

**Comment by df-sh** [comment]
Rackspace developer account free for a year for one server, costs around 30 -50 per month after that
AWS free tier free for a year as well, costs a little under 10 bucks for t2 micro server after that.
DO, 5-10USD per month
Linode havent tried
Google, had terrible time setting it up, havent gotten it running so no clue on cost.

For speeds from china
Rackspace HK has been the champ

**Comment by denoza** [comment]
@fghorow I was just wondering that BudgetVM's price is good. So do you mean that for the OpenVZ config (25$/year) streisand script doesn't go beyond l2tp/ip sec and you can't use it to run other protocols from streisand (by which I mean that streisand setup simply fails)?

As for the providers, I have used only two.

Vultr LA is working pretty good for me from North of China. (5$/month)
DO has been causing problems lately, but may be it is my ISP issue. (5$/month)

**Comment by fghorow** [comment]
@denoza Sorry if I was unclear. 

First off, I am not using streisand from China, so my set of priorities might not match yours.

L2TP/IPsec traffic *failed* to get through to my OpenVZ VM even though it appears that the VM's firewall was set up correctly by the streisand scripts. The host BudgetVM claimed that they do not firewall their machines, and I see no reason to doubt that. The L2TP/IPsec traffic failure is consistent with the claim in this link <https://www.lowendtalk.com/discussion/11171/implement-of-l2tp-in-an-openvz-container> that L2TP/IPsec and OpenVZ do not play well together.

The only other method I tested on the OpenVZ machine was the AnyConnect technique, and that worked.  I did not test any of the other techniques on OpenVZ. 

As soon as I removed my OpenVZ based VM, and replaced it with a (slightly more expensive) Xen based  machine, the L2TP/IPsec configuration set up by streisand started working perfectly. I have also tested the AnyConnect and OpenVPN based VPNs and those too appear to be working fine.

I can not comment on any of the other configs.

Bottom line: L2TP/IPsec and OpenVZ do not appear to play well together, at least in the version of OpenVZ being run by BudgetVM.

Hope this helps!

**Comment by unstatusthequo** [comment]
I've had great luck with DO and Vultr. Vultr is also **doubling** any amount you prepay with. So for example, I just prepaid $100 and have $200 in credit that I can use in the next 12 months.

**Comment by plaindocs** [comment]
Even just a $5 dollar month estimate for "normal" usage would be helpful.

**Comment by andreagrax** [comment]
I'm actually using the 1 euro vps from arubacloud
1 processor
1 gb ram
20 gb ssd
2 tb/month traffic

Mainly openvpn when I'm traveling and it's quite good for my usage , but
YMMV

Andrea

**Comment by TheNomad11** [comment]
I have been using the cheapest KVM VPS from Signaltransmitter.de - for only 1,50 Euro/month. 512MB RAM is more than enough, 1 processor. No issues so far. Streisand uses hardly any ressources

**Comment by tnguyen14** [comment]
@unstatusthequo do you have any instruction on how to set up on Vultr?

**Comment by cpu** [comment]
I don't think Streisand should make recommendations for specific cloud providers at this point. I think a user savvy enough to manage a Streisand installation can likely compare cloud providers themselves. This issue has diverged from the original topic to be a wiki-like document of individuals listing their own provider choices. I don't think this repository or a github issue is the correct place for that so I'm closing this issue. Thanks!
